src/main/java/com/exadel/discount/entity/User.java

```diff
@@ -0,0 +1,72 @@
+package com.exadel.discount.entity;
+
+import com.exadel.discount.config.EnumPostgresSQLType;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+import com.exadel.discount.model.security.Role;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Entity
+@Table(name = "users")
+@Data
+@NoArgsConstructor
+@TypeDef(
+        name = "user_role",
+        typeClass = EnumPostgresSQLType.class
+)
+public class User {
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(generator = "UUID")
+    @GenericGenerator(
+            name = "UUID",
+            strategy = "org.hibernate.id.UUIDGenerator"
+    )
+    UUID id;
+    @Column(name = "first_name", length = 50, nullable = false)
+    private String firstName;
+    @Column(name = "last_name", length = 50, nullable = false)
+    private String lastName;
+    @Column(name = "phone", length = 50, nullable = false)
+    private String phone;
+    @Column(name = "email", length = 50, nullable = false)
+    private String email;
+    @Column(name = "login", length = 50, nullable = false)
+    private String login;
+    @Column(name = "password", length = 225, nullable = false)
+    private String password;
+
+    @Enumerated(EnumType.STRING)
+    @Column(columnDefinition = "role")
+    @Type(type = "user_role")
```

Comment:
If you are using just enum with no fields inside there is a possibility to use  only   @Enumerated(EnumType.STRING)
no need to create own type. it's actually redundant

Reply:
It's needed for coordinate Role type in Model  with user_role Type in Liquibase